staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -222,7 +220,7 @@ func (r *Request) Resource(resource string) *Request {
 // or defaults to the stub implementation if nil is provided
 func (r *Request) BackOff(manager BackoffManager) *Request {
 	if manager == nil {
-		r.backoff = &NoBackoff{}
+		// bad input, so we will not change the current backoff
```

Comment:
when a user call `request.BackOff` this implies that the user wants to specify a custom `Backoff`, and passing `nil` is a programmer error. 
Ideally I think we should set the `Request.err` to "can't set a nil Backoff" and render the request object not usable. But this will also be a breaking change since some folks may be using `request.BackOff(nil)` as a way to reset the underlying backoff manager for the request.

we have these options:
- A: do nothing, it's a programmer error (not backward compatible)
- B: set the backoff to default `r.backoff = defaultBackoffManager()` - we are making a choice for the user here when he/she asked for a different thing maybe.
- C: we can go back to `r.backoff = &NoBackoff{}`, this retains old behavior, what do you think?

Reply:
I'm always vote for being conservative and keep compatibility unless is clear a bug, so I would say C, but let's wait for wojtek or jordan

Reply:
OK - I agree with Antonio - let's go with (C) and retain the old behavior.
It's something that we will be able to change later if needed independently, so let' try to be consiervative with this PR and we can potentially discuss that as a follow up.